onnxscript/rewriter/generic_pattern.py

```diff
@@ -297,20 +299,33 @@ def _match_backward(
             )
             return self.none(starting_node, inspect.currentframe().f_lineno)
 
-        for graph_input, pattern_input in zip(graph_node.inputs, pattern_node.inputs):
-            if len(list(graph_input.uses())) != len(list(pattern_input.uses())):
-                self._hint(
-                    "BACKWARD: one input is used outside the pattern",
-                    "-- pattern",
-                    pattern_node,
-                    "-- model",
-                    graph_node,
-                )
-                return self.none(starting_node, inspect.currentframe().f_lineno)
+        has_predecessor_in_pattern = any(
+            pattern_value.producer() is not None for pattern_value in pattern_node.inputs
+        )
+
+        if has_predecessor_in_pattern:
+            for graph_input, pattern_input in zip(graph_node.inputs, pattern_node.inputs):
+                if len(graph_input.uses()) != len(pattern_input.uses()):
```

Comment:
Instead of the has_predecessor_in_pattern check above, I think we should do the check here: add `if pattern_input.producer() is not None` here ... that will handle the case where some inputs have producer and others don't.

Reply:
changed